src/goto-instrument/insert_final_assert_false.cpp

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************\
+
+Module: Insert assert(false) at end of entry function
+
+Author: Nathan Chong, Kareem Khazem
+
+\*******************************************************************/
+
+/// \file
+/// Insert final assert false into a function
+
+#include "insert_final_assert_false.h"
+
+#include <goto-programs/goto_model.h>
+#include <util/irep.h>
+
+#include <iterator>
+#include <list>
+
+insert_final_assert_falset::insert_final_assert_falset(
+  message_handlert &_message_handler)
+  : messaget(_message_handler)
+{
+}
+
+bool insert_final_assert_falset::
+operator()(goto_modelt &goto_model, const std::string &function_to_instrument)
+{
+  const irep_idt entry(function_to_instrument);
+  auto it = goto_model.goto_functions.function_map.find(entry);
+  if(it == goto_model.goto_functions.function_map.end())
+  {
+    error() << "insert-final-assert-false: could not find function "
+            << "'" << function_to_instrument << "'" << eom;
+    return true;
+  }
+  goto_programt &entry_function = (it->second).body;
+  goto_programt::instructionst &instructions = entry_function.instructions;
+  auto instr_it = instructions.end();
+  auto last_instruction = std::prev(instr_it);
+  DATA_INVARIANT(
+    last_instruction->is_end_function(),
+    "last instruction in function should be END_FUNCTION");
+  last_instruction->turn_into_skip();
+  goto_programt::targett assert_target = entry_function.add_instruction();
+  exprt false_expr;
```

Comment:
The sequence of overwriting, adding, and then producing a new end_function instruction could be shortened by simply saying
`entry_function.insert_before_swap(last_instruction);
*last_instruction=goto_programt::make_assertion(false_exprt());
`

Reply:
Done.